sdk/keyvault/keyvault-secrets/test/internal/challengeAuthenticationCallbacks.spec.ts

```diff
@@ -165,6 +165,21 @@ describe("Challenge based authentication tests", function () {
       );
     });
 
+    it("does not throw if the resource URI matches the request", async () => {
+      await challengeCallbacks.authorizeRequestOnChallenge!({
+        getAccessToken: () => Promise.resolve(null),
+        request: createPipelineRequest({ url: "https://myvault.vault.azure.net" }),
```

Comment:
This is good, too, but I meant checking that something like "myvault.azure.net" does *not* match because you check for ".vault.azure.net".

Reply:
Oh! Wasn't totally sure. That makes sense.